command/agent/http.go

```diff
@@ -87,6 +88,7 @@ func NewHTTPServers(agent *Agent, config *Config, logOutput io.Writer) ([]*HTTPS
 			logger:   log.New(logOutput, "", log.LstdFlags),
 			uiDir:    config.UiDir,
 			addr:     httpAddr.String(),
+			prefix:   "https",
```

Comment:
OK yeah looks like we can do this easy enough in by using the passed in `http.Request`. Amended and re-squashed, what do you think of this? Do you think this addresses it now? Not too sure if the constants are necessary here given the smaller scope now.